### DISCUSSION TITLE
Slow response time in game service

### DISCUSSION THREAD

**Discussion by schvila** [dotnet/yarp#2213] [Q&A] [answered]
We are porting our gaming solution from IIS URL rewrite to YARP and we are facing 15% slower response time in YARP. Each game and we have 100 games is using a different port, and there are game support services as well. We are hosting stage, dev, and production services as well. All game and support services are Kestrel Minimal Web API .NET 7 Core Rest services. 
Is there any way to improve response time? We expected YARP should be faster than IIS URL rewrite.

**Comment by Tratcher** [comment]
Slower in what regard, latency, time to first response byte, time until last response byte, etc.? Can you give specific examples?

**Reply by karelz** [reply]
@schvila will you be able to provide more info to make it actionable?

**Comment by schvila** [comment]
We are using jmeter to emulate 200 users access and average execution of requests was slower. We discovered services are running faster once they are running under IIS and proxying using YARP. Probably there is some optimization under IIS + some benefits like stopping unused services. We have to stay with IIS for now.

**Reply by karelz** [reply, marked as answer]
Team discussion: We think this is most likely Kestrel configuration problem. We expect Kestrel to be faster IIS in most scenarios, including this one.
Either way, out of scope of YARP.
